edgar_tool/utils.py

```diff
@@ -35,3 +35,50 @@ def safe_get(d: Dict, *keys) -> Any:
 
 def unpack_singleton_list(l: Optional[List]) -> Union[str, List[str]]:
     return l if (l is None) or (len(l) != 1) else l[0]
+
+def parse_location_input(location_input: str | tuple | None) -> str | None:
+    """
+    Handles text search input for --peo_in or --inc_in.
+
+    This function processes the input to ensure it is in an acceptable format 
+    for location searches. Because CLI input like --peo_in "NY, OH" yields
+    python value ('NY','OH'), this function supports single or multiple locations 
+    provided as a string or a tuple. If the input is a tuple, it converts the tuple 
+    to a comma-separated string. It also removes any whitespace from the output 
+    string to prevent errors during further processing. Also validates that all
+    provided location codes are in the TEXT_SEARCH_LOCATIONS_MAPPING and prints
+    the list of acceptable codes if not.
+
+    Parameters:
+    location_input (str | tuple | None): The input location(s) to be parsed. 
+        It can be a single location as a string, multiple locations as a tuple 
+        of strings, or None.
+
+    Returns:
+    str: A string representation of the location(s) with no whitespace.
+
+    Raises:
+    ValueError: If the input is not a string, tuple, or None, or if any location 
+        in the input is not in the TEXT_SEARCH_LOCATIONS_MAPPING.
+    """
+
+    if not isinstance(location_input, (str, tuple, type(None))):
+        raise ValueError(
+            f'peo_in and inc_in must use format like "NY" or "NY,OH,etc" '
+            f'and be one of {TEXT_SEARCH_LOCATIONS_MAPPING}'
+        )
+
+    if isinstance(location_input, tuple):
+        for value in location_input:
+            if value not in TEXT_SEARCH_LOCATIONS_MAPPING:
+                raise ValueError(f"{value} not in {TEXT_SEARCH_LOCATIONS_MAPPING}")
+        location_input = ','.join(location_input)
+
+    elif isinstance(location_input, str):
+        if location_input not in TEXT_SEARCH_LOCATIONS_MAPPING:
+            raise ValueError(f"{location_input} not in {TEXT_SEARCH_LOCATIONS_MAPPING}")
```

Comment:
It would be great if we could support both the short codes and the long names as in the `--inc_in "Mexico"` example in the issue.

So a user should be able to provide `--inc_in "Mexico"` or `--inc_in "O5"` and achieve the same result.

Some challenges could include the comma and trailing space in some of the entries: `"M4": "Korea, Democratic People's Republic of "`

What do you think about this?

Reply:
Hi @GalenReich I can implement that :) I'll revise the PR in the next few days.